app/assets/stylesheets/functions/_grid-settings.scss

```diff
@@ -0,0 +1,12 @@
+// Conveniently retrieves grid-settings from the `$grid-settings` map
+// Update the grid-settings map by passing a pair in to the function as follows:
+//
+// `$grid-settings: grid-settings(key, value);`
+
+@function grid-settings($key, $value: false) {
```

Comment:
I think this should be broken into two functions: one to retrieve and one to update.

Right now, throughout the codebase, you see `grid-settings()` and it’s not super clear what it is doing, which can be two vastly different things.